update required HIP version to 3.3.0

psychocoderHPC opened the pull request:
Update to the latest HIP release. Even with this release alpaka is not 100% usable with HIP because there are still bugs in HIP those will be fixed with the next HIP release [ref](https://github.com/ROCm-Developer-Tools/HIP/issues/1982)

- fix wrong used rocrand branch
- expose HIP to CMake via environment variable `CMAKE_PREFIX_PATH`
- define `HIP_PLATFORM` and `HIP_RUNTIME` before install HIP

I hope this update is also solving the HIP CI issues in #1004

BenjaminW3 commented on .travis.yml:
```diff
@@ -234,9 +234,9 @@ matrix:
+      env: ALPAKA_CI_DOCKER_BASE_IMAGE_NAME=ubuntu:16.04 CXX=g++     CC=gcc   ALPAKA_CI_GCC_VER=5       CMAKE_BUILD_TYPE=Debug   ALPAKA_CI_STDLIB=libstdc++ ALPAKA_CI_BOOST_BRANCH=boost-1.67.0 ALPAKA_CI_CMAKE_VER=3.15.7 ALPAKA_ACC_GPU_HIP_ENABLE=ON ALPAKA_ACC_GPU_HIP_ONLY_MODE=ON ALPAKA_CI_HIP_BRANCH="rocm-3.3.0" ALPAKA_HIP_PLATFORM=nvcc ALPAKA_CUDA_ARCH="30;35" ALPAKA_CUDA_VERSION=9.2 ALPAKA_CUDA ...
     - name: HIP(nvcc9.2) + gcc-5 Release ALPAKA_ACC_GPU_HIP_ONLY_MODE
-      env: ALPAKA_CI_DOCKER_BASE_IMAGE_NAME=ubuntu:16.04 CXX=g++     CC=gcc   ALPAKA_CI_GCC_VER=5       CMAKE_BUILD_TYPE=Release ALPAKA_CI_STDLIB=libstdc++ ALPAKA_CI_BOOST_BRANCH=boost-1.71.0 ALPAKA_CI_CMAKE_VER=3.17.1 ALPAKA_ACC_GPU_HIP_ENABLE=ON ALPAKA_ACC_GPU_HIP_ONLY_MODE=ON ALPAKA_CI_HIP_BRANCH="roc-3.1.0" ALPAKA_HIP_PLATFORM=nvcc ALPAKA_CUDA_ARCH="30;35" ALPAKA_CUDA_VERSION=9.2 ALPAKA_CUDA_ ...
+      env: ALPAKA_CI_DOCKER_BASE_IMAGE_NAME=ubuntu:16.04 CXX=g++     CC=gcc   ALPAKA_CI_GCC_VER=5       CMAKE_BUILD_TYPE=Release ALPAKA_CI_STDLIB=libstdc++ ALPAKA_CI_BOOST_BRANCH=boost-1.71.0 ALPAKA_CI_CMAKE_VER=3.17.1 ALPAKA_ACC_GPU_HIP_ENABLE=ON ALPAKA_ACC_GPU_HIP_ONLY_MODE=ON ALPAKA_CI_HIP_BRANCH="rocm-3.2.0" ALPAKA_HIP_PLATFORM=nvcc ALPAKA_CUDA_ARCH="30;35" ALPAKA_CUDA_VERSION=9.2 ALPAKA_CUDA ...
```
Does not seem to find `rocm-3.2.0`

psychocoderHPC replied:
my fault. There is not 3.2.0 only 3.3.0. They jumped over 3.2.0.

BenjaminW3 commented:
It still fails with the HIP config issues also seen on other Pull requests. I have no clue where this is coming from now.

psychocoderHPC commented:
> It still fails with the HIP config issues also seen on other Pull requests. I have no clue where this is coming from now.

maybe an issue with the used CMake version?

psychocoderHPC commented:
Hip is only shipping there own `FindHip.cmake` https://travis-ci.com/github/alpaka-group/alpaka/jobs/337246455#L2552 
Maybe this produces the error. I am wondering why the second run is not showing this issue

BenjaminW3 commented:
The last merged PRs did not change anything around there and it still worked for them. We explicitly specify the CMake patch release so it should not have changed but who knows?

BenjaminW3 commented:
I deleted all the cached docker images in travis. Maybe this helps.

psychocoderHPC commented:
Note: I stopped all CI tests for this PR expect the HIP tests. No need to block the CI for hours to test HIP only.

psychocoderHPC commented:
IMO it is an issue by HIP https://github.com/ROCm-Developer-Tools/HIP/blob/2a03c8da1a7975f6470fbdd3f5129e46692c0d3e/CMakeLists.txt#L433-L466
I am not sure why it worked before but `hip-config.cmake` is only installed if you are on a hcc platform.

It looks like it is fixed in the dev branch https://github.com/ROCm-Developer-Tools/HIP/blob/dabba8a3678f8cf57994feaa5dfe5591621fb6ce/CMakeLists.txt#L461-L487 by https://github.com/ROCm-Developer-Tools/HIP/pull/2010

BenjaminW3 commented:
@psychocoderHPC Do you think this can get fixed soon or may we work around the issue in the mean time? We could disable the merge barrier for merging the open PRs and disable the HIP build for now.

sbastrakov commented:
I would agree with @BenjaminW3 , if we can't make HIP builds work at the moment, could temporarily disable them.

psychocoderHPC commented:
> @psychocoderHPC Do you think this can get fixed soon or may we work around the issue in the mean time? We could disable the merge barrier for merging the open PRs and disable the HIP build for now.

Let me look today into it. We had vacation last Friday so I was not able to look into it.

psychocoderHPC committed:
fix HIP and update to 3.3.0

- fix wrong used rocrand branch
- expose HIP to CMake via environment variable `CMAKE_PREFIX_PATH`
- define `HIP_PLATFORM` and `HIP_RUNTIME` before install HIP
- update to HIP 3.3.0

psychocoderHPC commented:
HIP build are now passing. Let us wait for the rest of the CI and then rebase all other PR's after this is merged.
